rpc/core/src/convert/notification.rs

```diff
@@ -1,47 +1,41 @@
-use std::sync::Arc;
+use std::ops::Deref;
 
-use crate::{notify::collector::ArcConvert, BlockAddedNotification, NewBlockTemplateNotification, Notification};
-use consensus_core::notify as consensus_notify;
+use crate::{BlockAddedNotification, NewBlockTemplateNotification, Notification};
+use event_processor::notify as event_notify;
 
 // ----------------------------------------------------------------------------
 // consensus_core to rpc_core
 // ----------------------------------------------------------------------------
 
-impl From<&consensus_notify::Notification> for Notification {
-    fn from(item: &consensus_notify::Notification) -> Self {
+impl From<event_notify::Notification> for Notification {
+    fn from(item: event_notify::Notification) -> Self {
         match item {
-            consensus_notify::Notification::BlockAdded(msg) => Notification::BlockAdded(msg.into()),
-            consensus_notify::Notification::NewBlockTemplate(msg) => Notification::NewBlockTemplate(msg.into()),
+            event_notify::Notification::BlockAdded(msg) => Notification::BlockAdded((*msg.deref()).clone().into()),
+            event_notify::Notification::NewBlockTemplate(msg) => Notification::NewBlockTemplate(msg.into()),
+            _ => todo!(),
         }
     }
 }
 
-impl From<&consensus_notify::BlockAddedNotification> for BlockAddedNotification {
-    fn from(item: &consensus_notify::BlockAddedNotification) -> Self {
+impl From<event_notify::BlockAddedNotification> for BlockAddedNotification {
+    fn from(item: event_notify::BlockAddedNotification) -> Self {
         Self { block: (&item.block).into() }
     }
 }
 
-impl From<&consensus_notify::NewBlockTemplateNotification> for NewBlockTemplateNotification {
-    fn from(_: &consensus_notify::NewBlockTemplateNotification) -> Self {
+impl From<event_notify::NewBlockTemplateNotification> for NewBlockTemplateNotification {
+    fn from(_: event_notify::NewBlockTemplateNotification) -> Self {
         Self {}
     }
 }
 
-/// Pseudo conversion from Arc<Notification> to Arc<Notification>.
-/// This is basically a clone() op.
-impl From<ArcConvert<Notification>> for Arc<Notification> {
-    fn from(item: ArcConvert<Notification>) -> Self {
-        (*item).clone()
-    }
-}
-
-impl From<ArcConvert<consensus_notify::Notification>> for Arc<Notification> {
-    fn from(item: ArcConvert<consensus_notify::Notification>) -> Self {
-        Arc::new((&**item).into())
-    }
-}
-
 // ----------------------------------------------------------------------------
 // rpc_core to consensus_core
 // ----------------------------------------------------------------------------
+
+impl From<Notification> for event_notify::Notification {
+    //TODO
```

Comment:
?

Reply:
Done. - Removed, falsely put `From<Notification>` as requirement of `T` in the collector, whereas only `Into<Notficication>` suffices.